components/VM_Arm/src/main.c

```diff
@@ -620,6 +620,29 @@ static void irq_handler(void *data, ps_irq_acknowledge_fn_t acknowledge_fn, void
     }
 }
 
+/* Default RAM initialization. Modules can overwrite this weak function with a
+ * custom initialization. e.g. to use memory shared with other components or
+ * mapped on demand.
+ */
+WEAK int init_ram_module(vm_t *vm)
```

Comment:
Prototype is not the same, `cookie` parameter is missing and return type is different. The latter is more serious; if somebody provided overloaded `init_ram_module` (like in vm-introspect), the return value is of `void` type and hence here the CPU register typically containing return value would be completely random.

While it can be argued that it is hard to see how overloaded versions of `init_ram_module` are able to provide non-NULL cookie, I think if we are really just inlining something, we shouldn't make other changes at the same time.